File: maven-projects/java/jmx/arduino-base/src/main/java/br/com/cams7/sisbarc/arduino/status/ArduinoEEPROM.java
/**
 * 
 */
package br.com.cams7.sisbarc.arduino.status;

/**
 * @author cams7
 *
 */
public abstract class ArduinoEEPROM extends ArduinoStatus {

	public static final byte THREAD_TIME_MAX = 0x07; // 7

	public static final byte DIGITAL_ACTION_EVENT_MAX = 0x1F; // 31
	public static final byte ANALOG_ACTION_EVENT_MAX = 0x7F; // 127

	private byte threadTime;
	private byte actionEvent;

	/**
	 * 
	 */
	public ArduinoEEPROM() {
		super();

		setThreadTime((byte) 0x00);
		setActionEvent((byte) 0x00);
	}

	/**
	 * @param status
	 * @param event
	 * @param pinType
	 * @param pin
	 */
	public ArduinoEEPROM(Status status, Event event, PinType pinType, byte pin,
			byte threadTime, byte actionEvent) {
		super(status, event, pinType, pin);

		setThreadTime(threadTime);
		setActionEvent(actionEvent);
	}

	public void changeCurrentValues(ArduinoEEPROM arduino) {
		super.changeCurrentValues(arduino);

		setThreadTime(arduino.getThreadTime());
		setActionEvent(arduino.getActionEvent());
	}

	@Override
	public String toString() {
		return this.getClass().getSimpleName() + "[transmitter = "
				+ getTransmitter() + ", status = " + getStatus() + ", event = "
				+ getEvent() + ", pinType = " + getPinType() + ", pin = "
				+ getPin() + ", threadTime = " + getThreadTime()
				+ ", actionEvent = " + getActionEvent() + "]";
	}

	public byte getThreadTime() {
		return threadTime;
	}

	public void setThreadTime(byte threadTime) {
		this.threadTime = threadTime;
	}

	public byte getActionEvent() {
		return actionEvent;
	}

	public void setActionEvent(byte actionEvent) {
		this.actionEvent = actionEvent;
	}

}

File: eclipse-workspace/arduino_sisbarc/src/ArduinoEEPROM.cpp
/*
 * ArduinoEEPROM.cpp
 *
 *  Created on: 31/01/2015
 *      Author: cams7
 */

#include "ArduinoEEPROM.h"

namespace SISBARC {

ArduinoEEPROM::ArduinoEEPROM() :
		ArduinoStatus(), EEPROMData() {
}

ArduinoEEPROM::ArduinoEEPROM(status statusValue, event eventValue,
		pin_type pinType, uint8_t pin, uint8_t threadTime, uint8_t actionEvent) :
		ArduinoStatus(statusValue, eventValue, pinType, pin), EEPROMData(
				threadTime, actionEvent) {
}

ArduinoEEPROM::ArduinoEEPROM(status statusValue, event eventValue,
		pin_type pinType, uint8_t pin, EEPROMData* data) :
		ArduinoStatus(statusValue, eventValue, pinType, pin) {
	setThreadTime(data->getThreadTime());
	setActionEvent(data->getActionEvent());
}

ArduinoEEPROM::~ArduinoEEPROM() {
}

ArduinoEEPROMRead::ArduinoEEPROMRead() :
		ArduinoEEPROM() {
	setEventValue(READ);
}

ArduinoEEPROMRead::ArduinoEEPROMRead(status statusValue, pin_type pinType,
		uint8_t pin, uint8_t threadTime, uint8_t actionEvent) :
		ArduinoEEPROM(statusValue, READ, pinType, pin, threadTime, actionEvent) {
}

ArduinoEEPROMRead::ArduinoEEPROMRead(status statusValue, pin_type pinType,
		uint8_t pin, EEPROMData* data) :
		ArduinoEEPROM(statusValue, READ, pinType, pin, data) {

}

ArduinoEEPROMRead::~ArduinoEEPROMRead() {
	// TODO Auto-generated destructor stub
}

ArduinoEEPROMWrite::ArduinoEEPROMWrite() :
		ArduinoEEPROM() {
	setEventValue(WRITE);
}

ArduinoEEPROMWrite::ArduinoEEPROMWrite(status statusValue, pin_type pinType,
		uint8_t pin, uint8_t threadTime, uint8_t actionEvent) :
		ArduinoEEPROM(statusValue, WRITE, pinType, pin, threadTime, actionEvent) {
}

ArduinoEEPROMWrite::ArduinoEEPROMWrite(status statusValue, pin_type pinType,
		uint8_t pin, EEPROMData* data) :
		ArduinoEEPROM(statusValue, WRITE, pinType, pin, data) {

}

ArduinoEEPROMWrite::~ArduinoEEPROMWrite() {
	// TODO Auto-generated destructor stub
}

} /* namespace SISBARC */
